### DISCUSSION TITLE
How to setup the environment variables for comple_MET_all.sh script

### DISCUSSION THREAD

**Discussion by technicastudies** [dtcenter/METplus#1470] [Installation] [answered]
As you have mentioned there are external libraries are required for MET and we have also "compile_MET_all.sh" script for installation of libraries but script expects that environment variables should be set before compile. May that why i have encounter the error. So, my question is that how we can set up the environment variables like TEST_BASE, COMPILER, etc.
I have used following command to compile the script. 
compile_MET_all.sh install_met_env.<machine_name> 
Please answer.
Thank you.

**Comment by j-opatz** [comment]
Hello,

And thank you for your question. It sounds like you're having some trouble knowing where to start for a MET installation.

For the environment variables you described (TEST_BASE, COMPILER, etc.), these should be set in the user's environment script. For bash users, it would be the .bashrc file; for csh users, it's the .cshrc file. 

So for example, if I'm a bash user and need to set my MET_TARBALL environment variable, I'd use

`export MET_TARBALL=met-10.1.0-beta5.tar.gz`

That way, when the compile_MET_all.sh script is run, it will utilize the preset MET_TARBALL variable and fill in "met-10.1.0-beta5.tar.gz" instead.

I'm going to provide you a link [here](https://dtcenter.org/events/2021/metplus-training-series/agenda-recordings) to the ongoing METplus training sessions we host. Under Session 11's dropdown, you'll find a wealth of information on compiling MET, including how to set the environment variables you're asking about. Hopefully this will help guide you through the installation process.

**Comment by jprestop** [comment, marked as answer]
Hi @technicastudies.  I work with @j-opatz and just wanted to follow up with some additional information.  These variables could also be set up in an installation configuration file with the name install_met_env.<machine_name>, filling in <machine_name> with the name of the machine you are working on since the configuration would likely be specific to that machine.  If you set up this file, you would then pass that to the compile_MET_all.sh script as you indicated:

> compile_MET_all.sh install_met_env.<machine_name>
(again, filling in  <machine_name> with the name of the machine you are working on)

The link @j-opatz under Session 11 would prove to be very useful.  You can also get a preview of that information on the on the [MET Download page](https://dtcenter.org/community-code/model-evaluation-tools-met/download) under the section titled "Sample Script For Compiling External Libraries And MET".  You can also view some sample install_met_env.<machine_name> scripts [here](https://github.com/dtcenter/MET/tree/develop/scripts/installation/config) in the MET repository.  

And to clarify, you mentioned:

> export MET_TARBALL=met-10.1.0-beta5.tar.gz

However, while the name of the MET tarball appears to be met-10.1.0-beta5.tar.gz, that is the link name.  If you copy the link address you'll see that the link address is:

> https://github.com/dtcenter/MET/releases/download/v10.1.0-beta5/met-10.1.0-beta5.20220114.tar.gz

and the tarball name is actually "met-10.1.0-beta5.20220114.tar.gz", which is the name you'd need to use in that location.  

However, we typically recommend that users install the latest stable release (in this case MET-10.0.1), which you can find here:
https://github.com/dtcenter/MET/releases/download/v10.0.1/met-10.0.1.20211201.tar.gz

If you did want to get the latest beta release, however, we did just release our beta6 release this week, which you can find here:
https://github.com/dtcenter/MET/releases/download/v10.1.0-beta6/met-10.1.0-beta6.20220302.tar.gz

Please let us know what other questions you may have.

**Comment by georgemccabe** [comment]
Hi @technicastudies , I am just following up on this discussion. Were you able to get your issues resolved?

**Reply by jprestop** [reply]
HI @technicastudies.  I just wanted to follow up again.  Were you able to get a successful compilation of MET?  Please let us know if you're still having trouble.  Thanks!

**Reply by jprestop** [reply]
Hi @technicastudies.  I'd like to mark this discussion as being answered and "lock it" to prevent future posts. That's how we encourage users to create new discussions for new questions. But I wanted to give you an opportunity to comment on it before doing so. Please feel free to select one of the responses as being the best answer to your original question. That'll help future users with similar questions find answers.
